arangod/Graph/PathManagement/PathValidatorOptions.h

```diff
@@ -81,27 +123,30 @@ class PathValidatorOptions {
 
   aql::FixedVarExpressionContext& getExpressionContext();
 
-  // @brief Only required for rolling upgrades 3.8 => 3.9
-  // If a graph is asked for the first vertex and that is filtered
+  // @brief If a graph is asked for the first vertex and that is filtered
   // it can be removed for 3.9 => nextVersion.
-  void compatibility38IncludeFirstVertex() {
-    _compatibility38IncludeFirstVertex = true;
+  void setBfsResultHasToIncludeFirstVertex() {
+    _bfsResultHasToIncludeFirstVertex = true;
   }
 
-  bool hasCompatibility38IncludeFirstVertex() const {
-    return _compatibility38IncludeFirstVertex;
+  bool bfsResultHasToIncludeFirstVertex() const {
+    return _bfsResultHasToIncludeFirstVertex;
   }
 
  private:
+  // Vertex expression section
   std::shared_ptr<aql::Expression> _allVerticesExpression;
   containers::FlatHashMap<uint64_t, std::shared_ptr<aql::Expression>>
       _vertexExpressionOnDepth;
-  aql::Variable const* _tmpVar;
-  arangodb::aql::FixedVarExpressionContext& _expressionCtx;
-
   std::vector<std::string> _allowedVertexCollections;
+  bool _bfsResultHasToIncludeFirstVertex = false;
+
+  // Prune section
+  std::shared_ptr<aql::PruneExpressionEvaluator> _pruneEvaluator;
+  std::shared_ptr<aql::PruneExpressionEvaluator> _postFilterEvaluator;
```

Comment:
I am not fully connived the Options are the right place to own the Evaluators.
The evaluators are a moving target, as they will always get new Context Data injected, basically rendering the options here not const for the Validator.
I would prefer the Validator to own the Evaluators, and the Options own the information to create new Evaluators if required.
Happy to do discuss this in a Call, and do it after merging this PR in a small refactoring.

Reply:
Created kanban card: https://arangodb.atlassian.net/browse/GORDO-1358